webapp/tests/Unit/Service/ScoreboardServiceTest.php

```diff
@@ -0,0 +1,148 @@
+<?php declare(strict_types=1);
+
+namespace App\Tests\Unit\Service;
+
+use App\Service\ScoreboardService;
+use Doctrine\Common\Collections\Order;
+use Exception;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+
+class ScoreboardServiceTest extends KernelTestCase
+{
```

Comment:
Not sure the tests here are very relevant as they test internal details. I'd focus on the behaviour tests at the end.

Reply:
They helped me uncover a small bug, so I think they were useful. If at one point they become tedious to update because how it is implemented internally frequently (I don't think so, but who knows) we can always delete them.